### ISSUE TITLE
Fix S6853 FP: Rule does not recognize asp-for="" for labels in MVC views

### ISSUE THREAD

**Issue by cbaerike** [SonarSource/sonar-dotnet#8849]
### Description

Rule S6853 in SonarCloud does not recognize .NET MVC syntax for labels in views. 

### Repro steps
Consider this code in a *.cshtml file:
```
<div class="form-group">
    <label asp-for="Foo"></label><br />
    <input asp-for="Foo" class="form-control" />
    <span asp-validation-for="Foo"></span>
</div>
```

And in the view model (*.cs file) this related code:
```
public class foobar {
    [Display(Name = "Foo Description")]
    public string Foo {get;set;}
}
```
This triggers rule S6853 because the label does not have a text label and a "for=" property that references a form element. However, .NET MVC renders this code as follows:
```
<div class="form-group">
    <label for="Foo">Foo Description</label><br>
    <input class="form-control" type="text" data-val="true" id="Foo" name="Foo" value="foobarbaz">
    <span class="valid-feedback" data-valmsg-for="Foo" data-valmsg-replace="true"></span>
</div>
```
Which is perfectly valid according to the rule.

### Expected behavior
Rule S6853 should not trigger when the label has an asp-for property and a there is a form element with an asp-for property that reference the same property name.

### Actual behavior
The rule does not recognize the .NET MVC code.

### Known workarounds
None.

### Related information

* C# in .NET 8
* Visual Studio 2022
* dotnet build via Azure DevOps pipeline
* SonarCloud

**Comment by dougwaldron** [comment]
Hey @sebastien-marichal,

Thanks for the tips. Unfortunately, none of them helped. The warning shows up in my IDE, so your first (and third?) suggestions wouldn't apply there. And I didn't want to remove *all* benefits of the HTML plugin as would be the case with your 2nd suggestion.

What ended up working for me was to create a new quality profile (at `https://sonarcloud.io/organizations/[org_key]/quality_profiles`) for HTML using the "Sonar way" as the parent. In the new profile, I disabled this rule (and also S6825, another FP for ASP.NET) and set it as the default:

![image](https://github.com/SonarSource/sonar-dotnet/assets/737326/bc36064a-de92-4206-9ca7-ce5f517620d4)

In connected mode, the rule changes synced after restarting my IDE.

**Comment by sebastien-marichal** [comment]
Hello @dougwaldron,

Thank you for sharing your solution, I should have mentioned it.

In case you copied your new profile from `Sonar way`, be aware, that any change or addition to the `Sonar way` profile won't be synced with your copy. You will need to update it manually.
